pkg/generators/clientgen/gen.go

```diff
@@ -376,7 +397,131 @@ func (g *Generator) generateSubInterfaces(ctx *genall.GenerationContext) error {
+					// +genclient:skipVerbs=get,update
+				*/
+
+				var additionalMethods []internal.AdditionalMethod
+				var skipVerbs []string
+				var onlyVerbs []string
+
+				genclientMarkers := info.Markers[genclientMarker.Name]
+
+				namespaceScoped := info.Markers.Get(nonNamespacedMarker.Name) == nil
+				noVerbs := info.Markers.Get(noVerbsMarker.Name) != nil
+				hasStatus := hasStatusSubresource(info)
+				readOnly := info.Markers.Get(readOnlyMarker.Name) != nil
+
+				// Extract values from skip verbs marker.
+				sVerbs := info.Markers.Get(skipVerbsMarker.Name)
+				if sVerbs != nil {
+					val, ok := sVerbs.(markers.RawArguments)
+					if !ok {
+						root.AddError(fmt.Errorf("marker defined in wrong format %q", skipVerbsMarker.Name))
+					}
+					skipVerbs = strings.Split(string(val), ",")
+				}
+
+				// Extract values from only verbs marker.
+				oVerbs := info.Markers.Get(onlyVerbsMarker.Name)
+				if oVerbs != nil {
+					val, ok := oVerbs.(markers.RawArguments)
+					if !ok {
+						root.AddError(fmt.Errorf("marker defined in wrong format %q", onlyVerbsMarker.Name))
+					}
+					onlyVerbs = append(onlyVerbs, strings.Split(string(val), ",")...)
+				}
+
+				if readOnly {
+					onlyVerbs = append(onlyVerbs, genutil.ReadonlyVerbs...)
+				}
+
+				for _, m := range genclientMarkers {
+					gm, ok := m.(genclient)
+					if !ok {
+						root.AddError(fmt.Errorf("marker defined in wrong format %q", genclientMarker.Name))
+					}
+
+					if gm.Method != nil {
+						additionalMethod := internal.AdditionalMethod{
+							Method:      gm.Method,
+							Verb:        gm.Verb,
+							Subresource: gm.Subresource,
+							Input:       gm.Input,
+							Result:      gm.Result,
+						}
+
+						if err := verifyAdditionalMethod(additionalMethod); err != nil {
+							root.AddError(err)
+						}
+						additionalMethods = append(additionalMethods, additionalMethod)
+					}
+
+					// parse onlyVerbs marker
+					if len(onlyVerbs) > 0 {
+						onlyVerbsSet := make(map[string]bool)
+						for _, o := range onlyVerbs {
+							onlyVerbsSet[o] = true
+						}
+
+						// Combine the verbs in only Verbs and skip Verbs list.
+						skippedverbsList := []string{}
+						for _, m := range genutil.SupportedVerbs {
+							skip := true
+							if _, ok := onlyVerbsSet[m]; ok {
+								skip = false
+							}
+
+							// Check for conflits if a verb is present in onlyVerb and also in a SkipVerb marker.
+							for _, v := range skippedverbsList {
+								if v == m {
+									root.AddError(fmt.Errorf("verb %q used both in genclient:skipVerbs and genclient:onlyVerbs", v))
+								}
+							}
+
+							if skip {
+								skippedverbsList = append(skippedverbsList, m)
+							}
+						}
+						skipVerbs = skippedverbsList
+					}
+				}
+
+				a, err := internal.NewAPI(
+					root,
+					info,
+					gv.PackageName,
+					string(version.Version),
+					g.applyConfigGenPkg,
+					namespaceScoped,
+					additionalMethods,
+					&importList,
```

Comment:
Passing slice pointers is definitely not idiomatic in Go and generally, I wouldn't expect a "constructor" to have side-effects.